analyzer/src/funTest/kotlin/managers/GoModFunTest.kt

```diff
@@ -32,13 +32,34 @@ import org.ossreviewtoolkit.utils.test.USER_DIR
 import org.ossreviewtoolkit.utils.test.patchExpectedResult
 
 class GoModFunTest : StringSpec() {
-    private val projectDir = File("src/funTest/assets/projects/synthetic/gomod").absoluteFile
-    private val vcsDir = VersionControlSystem.forDirectory(projectDir)!!
-    private val vcsUrl = vcsDir.getRemoteUrl()
-    private val vcsRevision = vcsDir.getRevision()
+    private val testDir = File("src/funTest/assets/projects/synthetic").absoluteFile
 
     init {
         "Project dependencies are detected correctly" {
+            val projectDir = testDir.resolve("gomod")
+            val vcsDir = VersionControlSystem.forDirectory(projectDir)!!
+            val vcsUrl = vcsDir.getRemoteUrl()
+            val vcsRevision = vcsDir.getRevision()
+            val definitionFile = projectDir.resolve("go.mod")
+            val vcsPath = vcsDir.getPathToRoot(projectDir)
+            val expectedResult = patchExpectedResult(
+                projectDir.parentFile.resolve("gomod-expected-output.yml"),
+                definitionFilePath = "$vcsPath/go.mod",
+                path = vcsPath,
+                revision = vcsRevision,
+                url = normalizeVcsUrl(vcsUrl)
+            )
+
+            val result = createGoMod().resolveSingleProject(definitionFile)
+
+            result.toYaml() shouldBe expectedResult
+        }
+
+        "Project dependencies are detected correctly if the main package does not contain any code" {
+            val projectDir = testDir.resolve("gomod-subpkg")
```

Comment:
Would it make sense to only assert that the main package is included? (or make it in any other way more obvious which package the "main" one is?

Reply:
Do you mean the test should not check the whole result, but only if the result contains this main package? Then it should probably also be renamed to something like "The main package is not removed if it does not contain any code".

Reply:
Hm, thinking again - I believe it's just fine as-is.